components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -2054,6 +2080,98 @@ public static GraphQLValidationResponseDTO validateGraphQLSchema(String filename
+                schema = AstPrinter.printAst(schemaDocument);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Unable to generate GraphQL schema from introspection."
+                                    + " Endpoint returned response code: "
+                                    + response.getStatusLine().getStatusCode());
+                }
+                throw new APIManagementException("Error occurred while generating GraphQL schema from introspection",
+                        ExceptionCodes.GENERATE_GRAPHQL_SCHEMA_FROM_INTROSPECTION_ERROR);
+            }
+        } catch (Exception e) {
+            if (log.isDebugEnabled()) {
+                log.debug(
+                        "Exception occurred while generating GraphQL schema from endpoint. Exception: "
+                                + e.getMessage(), e);
+            }
+            throw new APIManagementException("Error occurred while generating GraphQL schema from introspection",
+                    ExceptionCodes.GENERATE_GRAPHQL_SCHEMA_FROM_INTROSPECTION_ERROR);
+        }
+        return schema;
+    }
+
+    /**
+     * Retrieve the GraphQL schema from the specified URL.
+     *
+     * @param url The URL of the GraphQL schema to retrieve.
+     * @return The GraphQL schema as a string.
+     * @throws APIManagementException If an error occurs while retrieving the schema
+     */
+    public static String retrieveGraphQLSchemaFromURL(String url) throws APIManagementException {
+        String schema = null;
+        try {
+            URL urlObj = new URL(url);
+            HttpClient httpClient = APIUtil.getHttpClient(urlObj.getPort(), urlObj.getProtocol());
+            HttpGet httpGet = new HttpGet(url);
+            HttpResponse response = httpClient.execute(httpGet);
+
+            if (HttpStatus.SC_OK == response.getStatusLine().getStatusCode()) {
+                schema = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
+            } else {
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Unable to generate GraphQL schema from url." + " URL returned response code: "
+                                    + response.getStatusLine().getStatusCode());
+                    throw new APIManagementException("Error occurred while retrieving GraphQL schema from schema URL",
+                            ExceptionCodes.RETRIEVE_GRAPHQL_SCHEMA_FROM_URL_ERROR);
+                }
+            }
+        } catch (IOException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception occurred while generating GraphQL schema from url. Exception: " + e.getMessage(),
+                        e);
+                throw new APIManagementException("Error occurred while retrieving GraphQL schema from schema URL",
+                        ExceptionCodes.RETRIEVE_GRAPHQL_SCHEMA_FROM_URL_ERROR);
+            }
```

Comment:
Take the throw out of the if block. Also, change the log to an error log.

Reply:
Resolved by 8d382846d051b22c9138741c44a81fba6fb3e2eb